FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift

```diff
@@ -0,0 +1,275 @@
+import UIKit
+
+public class RecentlyFavoritedShelfCell: UICollectionViewCell {
+    public typealias ButtonAction = ((_ model: RecentlyFavoritedViewmodel, _ isFavorited: Bool) -> ())
+    
+    private static let titleHeight: CGFloat = 8
+    private static let priceTagHeight: CGFloat = 30
+    private let imageviewWidth: CGFloat = 128
+    private var defaultImage: UIImage? {
+        return UIImage(named: .noImage)
+    }
+    
+    public var buttonAction: ButtonAction?
+    private var model: RecentlyFavoritedViewmodel?
+    private var isFavorited: Bool = true {
+        didSet {
+            favoriteButton.isToggled = isFavorited
+        }
+    }
+    
+    private var task: URLSessionDataTask?
+    
+    private lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
```

Comment:
I think using RemoteImageView should be better here, then you do not need any `URLSessionDataTask` in this class and can use the image loading method provided